core/templates/pages/topics-and-skills-dashboard-page/topics-list/topics-list.component.ts

```diff
@@ -129,6 +135,57 @@ export class TopicsListComponent {
     }, () => {});
   }
 
+  /**
+   * @returns {boolean} Checks whether the SerialChapterLaunch feature
+   * flag is enabled.
+   */
+  isSerialChapterLaunchFeatureEnabled(): boolean {
+    return this.platformFeatureService.status.
+      SerialChapterLaunchCurriculumAdminView.isEnabled;
+  }
+
+  ngOnInit(): void {
+    if (this.isSerialChapterLaunchFeatureEnabled()) {
+      this.TOPIC_HEADINGS = [
+        'index', 'name', 'added_stories_count', 'published_stories_count',
+        'notifications', 'subtopic_count', 'skill_count', 'topic_status'];
+    } else {
+      this.TOPIC_HEADINGS = [
+        'index', 'name', 'canonical_story_count', 'subtopic_count',
+        'skill_count', 'topic_status'
+      ];
+    }
+  }
+
+  ngOnChanges(): void {
+    this.fullyPublishedStoriesCounts = [];
+    this.partiallyPublishedStoriesCounts = [];
+    this.totalChaptersInPartiallyPublishedStories = [];
+    this.publishedChaptersInPartiallyPublishedStories = [];
+
+    for (let i = 0; i < this.topicSummaries.length; i++) {
```

Comment:
Can we use `topicSummaries.forEach()` here and below?

Reply:
No, actually the index i is being used for other purposes as well.

Reply:
Ah, I see. Thank you for the confirmation.